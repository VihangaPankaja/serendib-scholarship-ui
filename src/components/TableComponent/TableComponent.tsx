import { render } from '@testing-library/react';
import React from 'react';
import DataTableComponent from '../DataTableComponent/DataTableComponent';
import ModalButton from '../../components/ModalButton/ModalButton';

const data = {
    student: {
        columns: [
            {
                key: 'id', name: 'ID', width: 10,
                formatter(props: any) {
                    return (
                        <ModalButton detail={props.row} id={props.row.id} type={props.type} />
                    );
                },
            },
            { key: 'name', name: 'Name' },
            { key: 'contactNo', name: 'Contact No' },
            { key: 'email', name: 'Email' },
            { key: 'university', name: 'Univeristy' },
            { key: 'course', name: 'Course of Study' },
            { key: 'startDate', name: 'Course Start Date' },
            { key: 'endDate', name: 'Course End Date' },
            { key: 'schoolEndDate', name: 'Schol. start Date' },
            { key: 'sponsor', name: 'Sponsor Name' }
        ],
        rows: [
            { id: 0, name: 'Example', contactNo: '', email: '', university: '', course: '', startDate: '', endDate: '', schoolEndDate: '', sponsor: '', files: [] },
            { id: 1, name: 'Demo', contactNo: '', email: '', university: '', course: '', startDate: '', endDate: '', schoolEndDate: '', sponsor: '', files: [] }
        ]
    },
    sponsor: {
        columns: [
            {
                key: 'id', name: 'ID', width: 10,
                formatter(props: any) {
                    return (
                        <ModalButton detail={props.row} id={props.row.id} type={props.type} />
                    );
                },
            },
            { key: 'name', name: 'Name' },
        ],
        rows: [
            { id: 0, name: 'Example' },
            { id: 1, name: 'Demo' }
        ]
    }
}

export default function TableComponent(props: any) {
    return (
        <DataTableComponent 
            columns={props.type === 'student' ? data.student.columns : data.sponsor.columns} 
            rows={props.type === 'student' ? data.student.rows : data.sponsor.rows} 
            type= {props.type}
        />
    );
}
