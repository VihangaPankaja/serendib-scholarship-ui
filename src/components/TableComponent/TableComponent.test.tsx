import { render } from "@testing-library/react";
import React from "react";
import { describe, it, expect } from "@jest/globals";
import TableComponent from "./TableComponent";

describe("TableComponent", () => {
  it("renders", () => {
    const wrapper = render(<TableComponent />);
    expect(wrapper.container).toMatchSnapshot();
  });
});