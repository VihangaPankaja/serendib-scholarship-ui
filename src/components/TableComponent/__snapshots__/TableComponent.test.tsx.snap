// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent renders 1`] = `
<div>
  <div
    aria-colcount="2"
    aria-rowcount="3"
    class="rdg r104f42s700-beta15"
    dir="ltr"
    role="grid"
    style="height: 100vh; grid-template-rows: 35px repeat(2, 35px); --rdg-header-row-height: 35px; --rdg-summary-row-height: 35px; --rdg-sign: 1; grid-template-columns: 80px 80px;"
  >
    <div
      aria-rowindex="1"
      class="rdg-header-row h197vzie700-beta15"
      role="row"
      style="--rdg-grid-row-start: 1;"
    >
      <div
        aria-colindex="1"
        aria-selected="false"
        class="rdg-cell c1wupbe700-beta15"
        role="columnheader"
        style="grid-column-start: 1; min-width: 80px;"
        tabindex="0"
      >
        ID
      </div>
      <div
        aria-colindex="2"
        aria-selected="false"
        class="rdg-cell c1wupbe700-beta15"
        role="columnheader"
        style="grid-column-start: 2; min-width: 80px;"
        tabindex="-1"
      >
        Name
      </div>
    </div>
    <div
      aria-rowindex="2"
      class="rdg-row r1otpg64700-beta15 rdg-row-even"
      role="row"
      style="--rdg-grid-row-start: 2; --rdg-row-height: 35px;"
    >
      <div
        aria-colindex="1"
        aria-readonly="true"
        aria-selected="false"
        class="rdg-cell c1wupbe700-beta15"
        role="gridcell"
        style="grid-column-start: 1;"
        tabindex="-1"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          0
        </button>
      </div>
      <div
        aria-colindex="2"
        aria-readonly="true"
        aria-selected="false"
        class="rdg-cell c1wupbe700-beta15"
        role="gridcell"
        style="grid-column-start: 2;"
        tabindex="-1"
      >
        Example
      </div>
    </div>
    <div
      aria-rowindex="3"
      class="rdg-row r1otpg64700-beta15 rdg-row-odd"
      role="row"
      style="--rdg-grid-row-start: 3; --rdg-row-height: 35px;"
    >
      <div
        aria-colindex="1"
        aria-readonly="true"
        aria-selected="false"
        class="rdg-cell c1wupbe700-beta15"
        role="gridcell"
        style="grid-column-start: 1;"
        tabindex="-1"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          1
        </button>
      </div>
      <div
        aria-colindex="2"
        aria-readonly="true"
        aria-selected="false"
        class="rdg-cell c1wupbe700-beta15"
        role="gridcell"
        style="grid-column-start: 2;"
        tabindex="-1"
      >
        Demo
      </div>
    </div>
  </div>
</div>
`;
